lib/cocoapods/target/build_settings.rb

```diff
@@ -871,7 +871,16 @@ def linker_names_from_libraries(libraries)
 
         # @return [Array<String>]
         define_build_settings_method :header_search_paths, :build_setting => true, :memoized => true, :sorted => true do
-          target.header_search_paths(:include_dependent_targets_for_test_spec => test_xcconfig? && non_library_spec, :include_dependent_targets_for_app_spec => app_xcconfig? && non_library_spec, :configuration => @configuration)
+          paths = target.header_search_paths(:include_dependent_targets_for_test_spec => test_xcconfig? && non_library_spec, :include_dependent_targets_for_app_spec => app_xcconfig? && non_library_spec, :configuration => @configuration)
+
+          dependent_vendored_xcframeworks = []
+          dependent_vendored_xcframeworks.concat vendored_xcframeworks
+          dependent_vendored_xcframeworks.concat dependent_targets.flat_map { |pt| pt.build_settings[@configuration].vendored_xcframeworks }
+          paths.concat dependent_vendored_xcframeworks.
+            select { |xcf| xcf.build_type.static_library? }.
+            map { |xcf| "#{BuildSettings.xcframework_intermediate_dir(xcf)}/Headers" }.
+            compact
+          paths
```

Comment:
Added this additional fix following the discussion on https://github.com/CocoaPods/cocoapods-integration-specs/pull/295#discussion_r491760000 — CocoaPods was including the XCF's Headers for aggregate targets, but not for the vending pod target itself (or targets which depend on it, including test targets).

Reply:
looks great - I think we were relying on `FRAMEWORK_SEARCH_PATHS`, though I am curious how any of the static lib implementation worked before without this. I know we had at least one example building with a static lib wrapped in an xcframework

Reply:
Thanks. If the calls to the static lib where happening in an app target, it would have been working by inheriting this from the aggregate target?